src/realm/query_engine.hpp

```diff
@@ -1418,42 +1418,45 @@ class StringNode<ContainsIns> : public StringNodeBase {
             m_ucase = std::move(*upper);
             m_lcase = std::move(*lower);
         }
-        
+
         if (v.size() == 0)
             return;
-        
+
         // Build a dictionary of char-to-last distances in the search string
         // (zero indicates that the char is not in needle)
         size_t last_char_pos = m_ucase.size()-1;
         for (size_t i = 0; i < last_char_pos; ++i) {
             // we never jump longer increments than 255 chars, even if needle is longer (to fit in one byte)
             uint8_t jump = last_char_pos-i < 255 ? static_cast<uint8_t>(last_char_pos-i) : 255;
-            
+
             unsigned char uc = m_ucase[i];
             unsigned char lc = m_lcase[i];
             m_charmap[uc] = jump;
             m_charmap[lc] = jump;
         }
 
     }
-    
+
     void init() override
     {
         clear_leaf_state();
-        
+
         m_dD = 100.0;
-        
+
         StringNodeBase::init();
     }
-    
-    
+
+
     size_t find_first_local(size_t start, size_t end) override
     {
         ContainsIns cond;
-        
+
         for (size_t s = start; s < end; ++s) {
             StringData t = get_string(s);
-            
+            // current (questionable?) behaviour is to return all results when querying for a null string
```

Comment:
The description at test/test_query.cpp:367 says `If A + B == B, then A is a prefix of B, and B is a suffix of A` which is the "axiom" on which all the reasoning about comparison is based...

Reply:
Thanks for the clarification, I'll update the comment.